core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -108,21 +104,24 @@
 
   private synchronized void appendCurrentBuffer(byte[] buf, int offset, int length)
       throws IOException {
-    if (mAvailableBytes < length) {
-      long bytesRequested = mTachyonFS.requestSpace(mBlockId, length - mAvailableBytes);
-      if (bytesRequested + mAvailableBytes >= length) {
-        mAvailableBytes += bytesRequested;
-      } else {
-        mCanWrite = false;
-        throw new IOException(String.format("No enough space on local worker: fileId(%d)"
-            + " blockId(%d) requestSize(%d)", mFile.mFileId, mBlockId, length - mAvailableBytes));
+    initializeLocalFile();
```

Comment:
If the client fails to initialize the local file, it can still call write but nothing is written and no exception. Do you want to reply on the first request to acutally writes the block and caches it? Since nothing is written by this BlockOutStream, do we still need call cacheBlock in the close() method?